walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,123 @@
+      ArrayList<TimeRange> eventList = new ArrayList<>();
+      for(Event event: events) {
+          Set<String> attendees = event.getAttendees();
+          for(String attendee: attendees) {
+              if(requestAttendees.contains(attendee)) {
+                eventList.add(event.getWhen());
+                break;
+              }
+          }
+      }
+      return eventList;
+  }
+
+  private ArrayList<TimeRange> createFlattenedEventList(Collection<Event> events, Collection<String> requestAttendees) {
+      ArrayList<TimeRange> eventList = createEventList(events, requestAttendees);
+      ArrayList<TimeRange> ret = new ArrayList<>();
+      Collections.sort(eventList, TimeRange.ORDER_BY_START);
+      int size = eventList.size();
+      if(size == 1) {
+          ret.add(eventList.get(0));
+          return ret;
+      }
+      for(int i = 0; i < size; i++) {
+          TimeRange curr = eventList.get(i);
+          TimeRange next = i + 1 == size ? null : eventList.get(i + 1);
+          if(next != null && curr.overlaps(next)) {
+              eventList.set(i + 1, TimeRange.fromStartEnd(curr.start(), Math.max(next.end(), curr.end()), false));
+          }
+          else {
+              ret.add(curr);
+          }
+      }
+      return ret;
+  }
+
+  private Collection<TimeRange> findMeetings(ArrayList<TimeRange> eventList, long requestDuration) {
+      ArrayList<TimeRange> ret = new ArrayList<>();
+      TimeRange firstRange = eventList.get(0);
+      if(eventList.get(0).equals(TimeRange.WHOLE_DAY)) {
+          return Arrays.asList();
+      }
+
+      if(eventList.size() == 1) {
+          handleSingleList(firstRange, requestDuration, ret);
+          return ret;
+      } 
+      else {
+          int size = eventList.size();
+          TimeRange lastRange = eventList.get(size - 1);
+          handleFirstListItem(firstRange, requestDuration, ret);
+          for(int i = 0; i < size - 1; i++) {
+              int start = eventList.get(i).end();
+              int end = eventList.get(i + 1).start();
+              if(end - start >= requestDuration) {
+                  ret.add(TimeRange.fromStartEnd(start, end, false));
+              }
+          }
+          handleLastListItem(lastRange, requestDuration, ret);
+      }
+      return ret;
+  }
+
+  private void handleSingleList(TimeRange range, long requestDuration, ArrayList<TimeRange> ret) {
+      int endOfDay = TimeRange.WHOLE_DAY.end();
+      int startOfDay = TimeRange.WHOLE_DAY.start();
+
+      if(range.start() == startOfDay && endOfDay - range.end() >= requestDuration) {
+          ret.add(TimeRange.fromStartEnd(range.end(), endOfDay, false));
+      }
+      else {
+          if(range.start() >= requestDuration) {
+              ret.add(TimeRange.fromStartEnd(startOfDay, range.start(), false));
+          }
+
+          if(endOfDay - range.end() >= requestDuration) {
+              ret.add(TimeRange.fromStartEnd(range.end(), endOfDay, false));
+          }
```

Comment:
I believe this case is the same as the case on line 106, except the case on line 106 is more specific (but ultimately returns the same result). You probably can get rid of the case on 106 (and remove the else block all together).